pkg/filter/filter.go

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright 2022 The KubeVela Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package filter
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"cuelang.org/go/cue"
+	"github.com/kubevela/pkg/cue/cuex"
+)
+
+// ApplyFilter applies the given filter to an object.
+func ApplyFilter(ctx context.Context, contextData map[string]interface{}, filter string) (bool, error) {
+	contextByte, err := json.Marshal(contextData)
```

Comment:
I think we can probably make some common functions in kubevela/pkg which accept template and contextData as input and generate cue.Value. It is commonly used across trigger/workflowstep/component/trait, etc.